### PR TITLE
[v5] Rename hooks in other packages

### PR DESCRIPTION
**Pull request by davidkpiano**

### CODE REVIEW

File: packages/xstate-solid/src/createActorRef.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { onCleanup } from 'solid-js';
+import { isServer } from 'solid-js/web';
+
+export function createActorRef<TBehavior extends AnyActorBehavior>(
+  behavior: TBehavior,
+  ...[options = {}]: RestParams<TBehavior>
+): ActorRefFrom<TBehavior> {
+  const machineWithConfig = behavior;
+
+  const service = interpret(machineWithConfig, options);
```
Comment (Andarist) [inline packages/xstate-solid/src/createActorRef.ts:13]:
```suggestion
  const actorRef = interpret(behavior, options);
```

File: packages/xstate-solid/src/useMachine.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import type { AnyStateMachine, StateFrom, InterpreterFrom, Prop } from 'xstate';
```
Comment (Andarist) [inline packages/xstate-solid/src/useMachine.ts:1]: likely this file should be removed (or rather renamed to `useActor`)

Comment (davidkpiano) [reply packages/xstate-solid/src/useMachine.ts]: I did and worked on making this more generic, but there are some type issues I need help with

**Comment by davidkpiano** [comment]
@Andarist Might need help with changesets here since I'm not getting the option to add changesets for the changed packages

File: packages/xstate-svelte/test/UseMachineNonPersistentSubcription.svelte [hunk 3]
```diff
@@ -21,13 +21,13 @@
...
 </script>
 
 <div>
   <button type="button" on:click={() => (visible = !visible)}>Toggle</button>
   {#if visible}
     <!-- inlined version of this doesn't unsubscribe from the store when the content gets hidden, so we need to keep this in a separate component  -->
-    <UseMachineNonPersistentSubcriptionChild {send} {state} />
-  {/if}
+    <UseMachineNonPersistentSubcriptionChild {send} {snapshot} />
+  {/if}useActor
```
Comment (Andarist) [inline packages/xstate-svelte/test/UseMachineNonPersistentSubcription.svelte:32]:
```suggestion
  {/if}
```

File: packages/xstate-vue/src/useMachine.ts [hunk 4]
```diff
@@ -1,59 +1,29 @@
...
-  send: Prop<TInterpreter, 'send'>;
-  service: TInterpreter;
-};
+import { useActor } from './useActor.ts';
+import { Ref } from 'vue';
 
+/**
+ * @deprecated Use `useActor(...)` instead.
+ */
 export function useMachine<TMachine extends AnyStateMachine>(
```
Comment (Andarist) [inline packages/xstate-vue/src/useMachine.ts:16]: I wonder how useful it really is to keep this around if its signature changed *anyway*. Right now, the second argument only accepts `InterpreterOptions` whereas previously it also accepted `InternalMachineImplementations`.

**Review by Andarist — CHANGES_REQUESTED** [review]
I juggled files a little bit more to clean up more places, rename them properly etc. I think this is now almost in a good state but I think that READMEs should also get adjusted here.

There are also signature inconsistencies between `useActor`, `useMachine` and `useActorRef` across packages but let's tackle that unification in a follow up PR.

**Comment by Andarist** [comment]
> @Andarist Might need help with changesets here since I'm not getting the option to add changesets for the changed packages

I can't reproduce this problem locally but remember that changeset files are just simple `.md` files. The CLI is a nice convenience (and it helps a lot!) but it's not a strict requirement to create those files. You can always just copy-paste some existing one and tweak it however you like.

**Comment by davidkpiano** [comment]
> I juggled files a little bit more to clean up more places, rename them properly etc. I think this is now almost in a good state but I think that READMEs should also get adjusted here.
> 
> There are also signature inconsistencies between `useActor`, `useMachine` and `useActorRef` across packages but let's tackle that unification in a follow up PR.

The source of truth for this documentation will actually be in the docs repo (https://github.com/statelyai/docs) - let's do this in a separate PR, since I want to make a good way of syncing those documentation pages with the README pages (maybe as simple as providing a link once the docs are updated).

**Comment by davidkpiano** [comment]
@Andarist I need help merging `next` into this one

**Comment by Andarist** [comment]
> The source of truth for this documentation will actually be in the docs repo (https://github.com/statelyai/docs) - let's do this in a separate PR, since I want to make a good way of syncing those documentation pages with the README pages (maybe as simple as providing a link once the docs are updated).

I feel like the syncing problem is separate. If you come up with a syncing solution, that PR should be focused on syncing - not on the docs updates. I feel like updating the code here without adjusting the user-facing API reference in the README when the API changes so much is something that shouldn't land. We don't have to make it perfect right now - just update the signatures and mention what arguments are required by them.

**Comment by davidkpiano** [comment]
Going to redo this since this is stale

**Pull request closed without merging**